src/main/java/design/MazeGenerator.java

```diff
@@ -0,0 +1,16 @@
+package design;
+
+import entities.DesignableMaze;
+
+public abstract class MazeGenerator {
+
+//    int numRows;
+//    int numCols;
+    DesignableMaze maze;
+
+    public MazeGenerator(DesignableMaze maze) {
+        this.maze = maze;
+    }
+
+    public abstract void generate();
```

Comment:
I feel like it might be good to have a few of the helper methods that you are guaranteed to use in here as well as abstract methods

Reply:
other maze generation algorithm classes would use different helper methods though